packages/typespec-vscode/test/extension/common/utils.ts

```diff
@@ -0,0 +1,112 @@
+import fs from "node:fs";
+import os from "node:os";
+import path, { join } from "node:path";
+import { ElectronApplication, Page, _electron } from "playwright";
+import { test as baseTest, inject } from "vitest";
+
+const __dirname = import.meta.dirname;
+export const projectRoot = path.resolve(__dirname, "../../../");
+export const imagesPath = path.resolve(projectRoot, "test/extension/images-linux");
+
+interface Context {
+  page: Page;
+  app: ElectronApplication;
+}
+
+type LaunchFixture = (options: {
+  extensionPath?: string;
+  workspacePath: string;
+  trace?: "on" | "off";
+}) => Promise<Context>;
+
+/**
+ * The core method of the test, this method is encapsulated.
+ * With the help of the `_electron` object, you can open a vscode and get the page object
+ */
+export const test = baseTest.extend<{
+  launch: LaunchFixture;
+}>({
+  launch: async ({ task }, use) => {
+    const teardowns: (() => Promise<void>)[] = [];
+
+    await use(async (options) => {
+      const executablePath = inject("executablePath");
+      const workspacePath = options.workspacePath;
+      let envOverrides = {};
+      const codePath = path.join(executablePath, "../bin");
+      envOverrides = {
+        PATH: `${codePath}${path.delimiter}${process.env.PATH}`,
+      };
+      const tempDir = await fs.promises.mkdtemp(path.join(os.tmpdir(), "typespec-automation"));
+
+      const app = await _electron.launch({
+        executablePath,
+        env: {
+          ...process.env,
+          ...envOverrides,
+        },
+        args: [
+          "--no-sandbox",
+          "--disable-gpu-sandbox",
+          "--disable-updates",
+          "--skip-welcome",
+          "--skip-release-notes",
+          "--disable-workspace-trust",
+          `--extensions-dir=${path.resolve(tempDir, "extensions")}`,
+          `--extensionDevelopmentPath=${path.resolve(projectRoot)}`,
+          `--user-data-dir=${path.resolve(tempDir, "user-data")}`,
+          `--folder-uri=file:${path.resolve(workspacePath)}`,
+        ].filter((v): v is string => !!v),
+      });
+      const page = await app.firstWindow();
+      const tracePath = join(projectRoot, "test-results", task.name, "trace.zip");
+      await page.context().tracing.start({ screenshots: true, snapshots: true, title: task.name });
```

Comment:
I have setup plawywright tracing which create a trace.zip which can be used to have a full replay of what happend. Please take a look and see if we still need all the delay and screenshots

Reply:
@timotheeguerin The `sleep` method in the `precontrastresult` function has also been replaced with Playwright's built-in `waitForSelector` method. Please review.